src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -144,17 +145,25 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
   }
 
   private async fetchSearchResultPreviews(suggestionText: string) {
-    const previousQueryOptions = this.queryController.getLastQuery();
-    previousQueryOptions.q = suggestionText;
-    previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
+    const query = this.buildQuery(suggestionText);
     this.logShowQuerySuggestPreview();
-    const results = await this.queryController.getEndpoint().search(previousQueryOptions);
+    const results = await this.queryController.getEndpoint().search(query);
     if (!results) {
       return [];
     }
     return this.buildResultsPreview(suggestionText, results);
   }
 
+  private buildQuery(searchQuery: string): IQuery {
+    const { searchHub, pipeline } = this.queryController.getLastQuery();
```

Comment:
Maybe I'll try to get more info from the last query:
- tab
- locale
- timezone
- context

Reply:
Yes, the tab is especially important to ensure the suggestions are scoped to the current tab.

Reply:
Alright, I'm adding those fields.